runtime/include/chpl-tasks.h

```diff
@@ -243,15 +243,38 @@ char* chpl_task_idToString(
                chpl_taskID_t); //Task ID
 
 //
-// Yield.
+// Yields the current task. If Quiescent State-Based Reclamation
+// is enabled, it will invoke a checkpoint which will invalidate
+// any references to any protected data.
 //
 void chpl_task_yield(void);
 
+//
+// Yields the current task. If Quiescent State-Based Reclamation
+// is enabled, it will invoke a checkpoint if the passed counter
+// is some multiple of CHPL_QSBR_ITERATIONS_PER_CHECKPOINT. This
+// should be invoked during tight loops which yield repeatedly.
+//
+void chpl_task_yield2(int);
```

Comment:
I am not the least bit fond of this name.  :-)
That said, I can't actually think of one I like better.  As an alternative, therefore, how about just adding an int argument to the original chpl_task_yield() that gives the tasking implementation some notion of what's going on?  I'm thinking that `arg < 0` would mean something like "this task has nothing to do; do whatever you like" and `arg >= 0` would mean "this task is in a tight wait loop and this is the Nth iteration; make calls as you deem appropriate".  Then on the inside, `chpl_task_yield()` could invoke the QSBR checkpoint if `arg < 0 || arg % CHPL_QSBR_ITERATIONS_PER_CHECKPOINT == 0`.  (It would actually work for now if we were to pass `0` for the usage I specified for `< 0` above, but I'm trying to think ahead to future needs and I believe it could be useful to separate the two cases.)

Reply:
I suppose I can modify the original `chpl_task_yield`, but it would significant increase the number of files being changed (any tests and modules using `chpl_task_yield` would have an edit, unless there is a way to force it to implicitly pass the new argument)

Reply:
Chapel does have optional arguments, e.g. `proc chpl_task_yield(i:int=0)`.
It's only tricky because we might need to rename the C function, or use the facility to have the Chapel compiler give it a different name.

Reply:
Hmm.  I'd rather retain the `chpl_task_yield()` name for the tasking layer interface and use something else  (`chpl_yield()` for example) as the module-code wrapper.  But that requires all the renaming in the Chapel realm @LouisJenkinsCS mentioned.  Unless one of you can think of something better let's just leave it at `chpl_task_yield2()` and maybe someday we'll clean it up.